### PR TITLE
Baseline cpptrace integration

### PR DESCRIPTION
**Pull request by jeremy-rifkin**
This PR gets rid of stack trace implementations in libassert and integrates [cpptrace](https://github.com/jeremy-rifkin/cpptrace)
This PR also eliminates Makefiles, improves CMake scripts, and does some general cleanup.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -142,3 +145,50 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
...
+    target_link_libraries(demo PRIVATE assert)
+    target_compile_options(
+      demo
+      PRIVATE
+      "-DASSERT_USE_MAGIC_ENUM"
+      "-DASSERT_FAIL=custom_fail"
+      "-DASSERT_LOWERCASE"
+    )
+
+    file(GLOB_RECURSE integration_sources tests/integration/*.cpp)
```
Comment (Morwenn) [inline CMakeLists.txt:165]: You might want `CONFIGURE_DEPENDS` here, you're already using CMake 3.12 so it should be available.

File: CMakeLists.txt [hunk 2]
```diff
@@ -142,3 +145,50 @@ if(NOT CMAKE_SKIP_INSTALL_RULES)
...
+      "-DASSERT_LOWERCASE"
+    )
+    add_test(
+      NAME integration
+      WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tests
+      COMMAND
+      python3 run-tests.py $<TARGET_FILE:integration> ${CMAKE_BUILD_TYPE} ${CMAKE_CXX_COMPILER_ID} ${CMAKE_CXX_STANDARD}
+    )
+
+    file(GLOB test_sources tests/unit/*.cpp)
```
Comment (Morwenn) [inline CMakeLists.txt:182]: Same here for `CONFIGURE_DEPENDS`.

File: CMakeLists.txt (hunk 1 above)
Comment (jeremy-rifkin) [reply CMakeLists.txt:165]: Thank you!

Comment (jeremy-rifkin) [reply CMakeLists.txt]: I went ahead and just removed the glob entirely

**Pull request merged**